tests/ui/iter_skip_next_unfixable.rs

```diff
@@ -0,0 +1,9 @@
+#![warn(clippy::iter_skip_next)]
+
+/// Checks implementation of `ITER_SKIP_NEXT` lint
+fn main() {
+    // fix #8128
+    let test_string = "1|1 2";
+    let sp = test_string.split('|').map(|s| s.trim());
+    let _: Vec<&str> = sp.skip(1).next().unwrap().split(' ').collect();
```

Comment:
Could you also add a test where the iterator comes from a function or closure argument to make sure that those patterns also work? A closure might be easier as it doesn't require a type annotations :). A check if `if let` would also be awesome :upside_down_face: